openoa/utils/downloader.py

```diff
@@ -330,6 +340,200 @@ def get_era5_monthly(
+    https://cds.climate.copernicus.eu/how-to-api
+
+    This function returns hourly ERA5 data from the "ERA5 hourly data on single levels from 1940 to
+    present" dataset at the nearest grid point to the provided coordinates. See further details
+    regarding the dataset at:
+    https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels.
+    U and V components of wind speed at 100 m, temperature at 2 m, and surface pressure are
+    downloaded here.
+
+    As well as returning the data as a dataframe, the data is also saved as a csv file with the
+    concatenated data. These are located in the "save_pathname" directory, with "save_filename"
+    prefix. This allows future loading without download from the CDS service.
+
+    Args:
+        lat(:obj:`float`): Latitude in WGS 84 spatial reference system (decimal degrees).
+        lon(:obj:`float`): Longitude in WGS 84 spatial reference system (decimal degrees).
+        save_pathname(:obj:`str` | :obj:`Path`): The path where the downloaded reanalysis data will
+            be saved.
+        save_filename(:obj:`str`): The file name used to save the downloaded reanalysis data.
+        start_date(:obj:`str`): The starting year, month, and day that data is downloaded for. This
+            should be provided as a string in the format "YYYY-MM-DD". Defaults to "2000-01-01".
+        end_date(:obj:`str`): The final year, month, and day that data is downloaded for. This should be
+            provided as a string in the format "YYYY-MM-DD". Defaults to current date. Note that data
+            may not be available yet for the most recent couple months.
+        calc_derived_vars (:obj:`bool`, optional): Boolean that specifies whether wind speed, wind
+            direction, and air density are computed from the downloaded reanalysis variables and
+            saved. Defaults to False.
+
+
+    Returns:
+        df(:obj:`dataframe`): A dataframe containing time series of the requested reanalysis
+            variables:
+            1. u_ms: the U component of wind speed at a height of 100 m in m/s.
+            2. v_ms: the V component of wind speed at a height of 100 m in m/s.
+            3. temperature_K: air temperature at a height of 2 m in Kelvin.
+            4. surf_pres_Pa: surface pressure in Pascals.
+
+    Raises:
+        ValueError: If the start_date is greater than the end_date.
+    """
+
+    logger.info("Please note access to ERA5 data requires registration")
+    logger.info("Please see: https://cds.climate.copernicus.eu/api-how-to")
+
+    # set up cds-api client
+    try:
+        c = cdsapi.Client()
+    except SSLError:
+        print("Skipping certificate verification")
+        c = cdsapi.Client(verify=False)  # verification error for self-signed certificate
+
+    # create save_pathname if it does not exist
+    save_pathname = Path(save_pathname).resolve()
+    if not save_pathname.exists():
+        save_pathname.mkdir()
```

Comment:
Do we need `parents=True` to create any missing directories higher up than the last folder?